<launch>
  <arg name="depth_image" default="/camera/depth/image_raw"/>
  <arg name="depth_info" default="/camera/depth/camera_info"/>
  <arg name="fixed_frame_id" default="odom"/>
  
  <arg name="profile_args" default="valgrind --tool=callgrind"/>
  
  <arg name="debug" default="false"/>
  
  <arg name="prefix" default="xterm -e gdb --args" if="$(arg debug)" />
  <arg name="prefix" default="" unless="$(arg debug)" />

  <arg name="debugproj" default="false"/>
  <arg name="projected_prefix" default="xterm -e gdb --args" if="$(arg debugproj)" />
  <arg name="projected_prefix" default="" unless="$(arg debugproj)" />
  
  <group ns="egocylinder">


    <node name="depth_to_egocylindrical_node" pkg="egocylindrical" type="egocylindrical_propagator_node" output="log" launch-prefix="$(arg prefix)">        
      <param name="image_in" value="$(arg depth_image)"/>
      <param name="info_in" value="$(arg depth_info)"/>
      <param name="points_out" value="data"/>
      <param name="filtered_points" value="filtered_points"/>
      <param name="fixed_frame_id" value="$(arg fixed_frame_id)"/>
    </node>

    <node name="egocylindrical_to_pointcloud" pkg="egocylindrical" type="point_cloud_node" output="log">
      <remap from="egocylindrical_points" to="data"/>
      <remap from="cylindrical" to="points"/>
    </node>
    
    <node name="egocylindrical_to_projected_pointcloud" pkg="egocylindrical" type="projected_point_cloud_node" output="log" launch-prefix="$(arg projected_prefix)">
      <remap from="egocylindrical_points" to="data"/>
    </node>
    
    <node name="egocylindrical_to_range_image" pkg="egocylindrical" type="range_image_node" output="log" >
      <remap from="egocylindrical_points" to="data"/>
      <param name="use_raw" value="False"/>
      <param name="image_topic" value="image"/>
    </node>
    
    <node name="egocylindrical_to_raw_range_image" pkg="egocylindrical" type="range_image_node" output="log" >
      <remap from="egocylindrical_points" to="data"/>
      <param name="use_raw" value="True"/>
      <param name="image_topic" value="image_raw"/>
    </node>
    
    <node name="range_image_inflator" pkg="egocylindrical" type="range_image_inflator_node" output="log" launch-prefix="xterm -e gdb -ex run --args">
      <remap from="egocylindrical_points" to="data"/>
      <remap from="range_image" to="image"/>
      <param name="image_topic" value="inflated_image"/>
    </node>
    
    <node name="inflated_range_converter" pkg="egocylindrical" type="range_image_converter_node" output="log">
      <remap from="data_out" to="inflated_data"/>
      <remap from="image_in" to="inflated_image"/>
      <remap from="info_in" to="data"/>
    </node>
    
    <node name="inflated_ec_ppoint_publisher" pkg="egocylindrical" type="point_cloud_node" output="log">
      <remap from="egocylindrical_points" to="inflated_data"/>
      <remap from="cylindrical" to="inflated_points"/>
    </node>
  
  </group>
  
</launch>

